### ISSUE TITLE
Deal with QWebSettings which are immutable after QWebPage construction

### ISSUE THREAD

**Issue by annulen** [annulen/webkit#69]
Some settings are immutable after QWebPage construction: QWebSettngs value may be freely changed by this changes have no effect. The only identified setting for now is localStoragePath, but there may be more (and others might become immutable in future).

For example, see WKWebView documentation [1], configuration is immutable and passed to constructor.

* We need to document that localStoragePath can only be set via global QWebSettngs before page is created
* We may want to add QWebPage/QWebView constructors with QWebSettings argument, or create separate class for such immutable preferences

[1] https://developer.apple.com/library/ios/documentation/WebKit/Reference/WKWebView_Ref/#//apple_ref/occ/instp/WKWebView/configuration

**Comment by annulen** [comment]
Actually there are 4 kinds of settings in WebKit: 
* Global (e.g., RuntimeEnabledFeatures)
* PageGroup level (AFAIK only caption display preferences currently)
* Page level immutable (may be emulated by setting respective global before Page is created)
* "Regular" Page level settings which can be changed freely

**Comment by annulen** [comment]
Now Page also cannot change it's SessionID at runtime, which makes it impossible to flip between private and non-private mode.

We could try to work around this by using `QWebPageAdapter::deletePage()` and re-creating Page with new configuration when certain properties are modified

**Comment by Emdek** [comment]
@annulen, just for Qt 5.x?
AFAIR QtWebEngine doesn't allow to flip it.

**Comment by annulen** [comment]
If you are implying profile topic, it can be introduced without API break. In that case we can just make QWebPage created with old way just use global settings

**Comment by Emdek** [comment]
@annulen, nah, just wondering if this would be workaround just for Qt 5.x or remain afterwards.

**Comment by annulen** [comment]
If we keep QWebSettings object mutable, and these attributes are not excluded, it can be expected by users that all attributes should be mutable

**Comment by annulen** [comment]
Note that re-creation of Page object requires reloading of page to make it "alive"
